### PR TITLE
refactorings to support testing with a kas-installer instance

### PR DESCRIPTION
**Pull request by shawkins**
also creating a more balanced test for partitions so that there's additional consumer groups

### CODE REVIEW

File: perf/src/main/java/org/bf2/performance/InstanceProfiler.java [hunk 1]
```diff
@@ -311,8 +317,8 @@ public static void main(String[] args) throws Exception {
     private void runLocalTest() throws Exception {
         Profile profile = LATENCY;
         sizeAndUpdateConfig();
-        deployIfNeeded("profile");
-        OMBDriver driver = createDriver(instanceBootstrap, profile);
+        deployIfNeeded();
+        updateDriver(profile);
 
         byte[] storeBytes = Base64.getDecoder().decode(kafkaProvisioner.getTlsConfig().getTrustStoreBase64());
         File store = new File("target", "listener.jks");
```
Comment (MikeEdgar) [inline perf/src/main/java/org/bf2/performance/InstanceProfiler.java:324]: It doesn't need to be fixed here, but when I used this method locally, I needed to replace the keystore load with first reading `ssl.truststore.location` from the `appServicesProperties`.

**Review by MikeEdgar — COMMENTED** [review]
Should we also remove the `compression.type=uncompressed` from the topic configs? It will be a certain problem when using newer versions of the topic policy plugin. Easy enough to patch it if necessary though.

I can take a bit of time tomorrow and spin up some clusters to try it.

File: perf/src/main/java/org/bf2/performance/InstanceProfiler.java (hunk 1 above)
Comment (shawkins) [reply perf/src/main/java/org/bf2/performance/InstanceProfiler.java:324]: Yes, I haven't ever run the local test execution against a kas-installer cluster.  This method is mostly for debugging purposes, so I'm fine with whatever changes here.

**Comment by shawkins** [comment]
> Should we also remove the compression.type=uncompressed from the topic configs?

That's fine too.  I had briefly been using the compression scenario to test the affects of storage bandwidth - compressed from the producer and uncompressed at the broker leading to very high EBS demand, but of course that will no longer be allowed.  OMB now has support for randomization for testing the compressed scenarios, which I haven't tried out yet.

**Comment by shawkins** [comment]
Updated with the explicit topic compression removed.

**Review by MikeEdgar — APPROVED** [review]
Haven't had a chance to try it myself yet, but all looks good. Something I'd like to look at in the future is skipping all the kafka-client setup when the intent is to use a local worker.

**Review by rareddy — APPROVED** [review]
lgtm

File: perf/src/main/java/org/bf2/performance/ManagedKafkaProvisioner.java [hunk 2]
```diff
@@ -520,18 +517,23 @@ ManagedKafkaDeployment deployCluster(String namespace, ManagedKafka managedKafka
...
+        Resource<ManagedKafka> mkResource = cluster.kubeClient()
+                .client()
+                .resources(ManagedKafka.class)
+                .inNamespace(KAFKA_NAMESPACE)
+                .withName(name);
+        try {
+            return mkResource.get();
+        } catch (KubernetesClientException e) {
+
+        }
```
[the remark this answers is not in the document]
Comment (shawkins) [reply perf/src/main/java/org/bf2/performance/ManagedKafkaProvisioner.java]: I'm not sure why that was added, removed now.

**Pull request merged**